rendering/glb.ksy

```diff
@@ -0,0 +1,64 @@
+meta:
+  id: glb
+  title: glTF binary format
+  file-extension: glb
+  endian: le
+  license: MIT
+  xref:
+    doc-ref: https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#binary-gltf-layout
+    mime: model/gltf-binary
+
+seq:
+  - id: header
+    type: header
+  - id: chunks
+    type: chunk
+    repeat: eos
+
+types:
+
+  header:
+    seq:
+      - id: magic
+        contents: 'glTF'
+      - id: version
+        type: u4
+      - id: length
+        type: u4
+
+  chunk:
+    seq:
+      - id: length
+        type: u4
+      - id: type
+        type: u4
+        enum: chunk_type
+      - id: data
+        size: length
+        type:
+          switch-on: type
+          cases:
+            'chunk_type::json': json
+            'chunk_type::bin': bin
+
+  json:
+    seq:
+      - id: data
+        type: str
+        encoding: UTF-8
+        size-eos: true
+        doc: |
+          This is where GLB deviates from being an elegant format.
+          To parse the rest of the file, you have to parse the JSON first.
+
+  bin:
+    seq:
```

Comment:
Looks unfinished.

Reply:
In what sense? A GLB file is made up of data chunks. The chunk can be either a JSON chunk or a binary chunk. This is a binary chunk. There's no way to know what sort of data you're looking at in here though, because, for that, you'd have to first parse the JSON to get the offsets out of there.

Reply:
Please look at https://github.com/kaitai-io/kaitai_struct_formats/pull/183/files#diff-ce5a8a6f65e08325535c534e8222ecbd9f53a9749ba538b04cf857b5b07a0795R89 and https://github.com/kaitai-io/kaitai_struct_formats/pull/213/files#diff-35cf895592a530eee852a5354aaa16e650100fdfddb1e81752f7e6d0ef036786R115 

for the ideas how one can try to solve the issue of having data in some another format.

Reply:
Neither of those appear to actually solve the issue, but rather push it out to a single-language solution, which doesn't really fit with the spirit of Kaitai anyway.

Reply:
>Neither of those appear to actually solve the issue, but rather push it out to a single-language solution which doesn't really fit with the spirit of Kaitai anyway.

I completely aggree.

>What would be nice is a native JSON type in Kaitai, so that we can then just traverse into the tree of JSON objects.

It would be nice to have KS extensible, because blessing JSON over other serialization formats, including custom ones, is incorrect. And the impl still will be language-dependent. The same way as runtimes are and KS itself is. All of this is just an abstraction layer around languages.

>So I guess one valid solution is, define a new format for JSON and import that format, so we can declare the type of this piece as json. And then say that implementing the JSON format is up to the reader. LOL

So a valid solution is finishing and landing interfaces proposal first, and then standardizing some interfaces and creating a standard library for each interface. See the discussion in #457.

Reply:
I wasn't saying that json would be the only format to implement, nor was I saying that json would be implemented by kaitai itself. I was just saying that we could implement a placeholder 'type' for any given unknown type, and then how that's implemented is up to the future. If the future happens to be some interface where the type is converted to something else using external code, that's fine. If someone figures out how to complete a kts file for json, that's fine too.